src/TextForEvent.tsx

```diff
@@ -467,7 +466,7 @@ function textForPowerEvent(event: MatrixEvent): () => string | null {
         }
         if (from === previousUserDefault && to === currentUserDefault) { return; }
         if (to !== from) {
-            const name = UserIdentifierCustomisations.getDisplayUserIdentifier(userId, { roomId: event.getRoomId() });
+            const name = getRoomMemberDisplayname(event, userId);
```

Comment:
This won't handle disambiguation properly, if there are 3 people in the room called `Alice` it'll say something like

> Changed the power level of Alice to 51, Alice to 90, Alice to 91.

Reply:
Good catch! I missed that in the other places in this file it's used as e.g. `ev.sender?.name || getRoomMemberDisplayname(ev)`. 🤦‍♂️

How about we change the implementation of `getRoomMemberDisplayname` to first try the `name` property?

```
diff --git a/src/TextForEvent.tsx b/src/TextForEvent.tsx
index b94f7361e0..6be8dee332 100644
--- a/src/TextForEvent.tsx
+++ b/src/TextForEvent.tsx
@@ -54,7 +54,7 @@ function getRoomMemberDisplayname(event: MatrixEvent, userId = event.getSender()
     const client = MatrixClientPeg.get();
     const roomId = event.getRoomId();
     const member = client.getRoom(roomId)?.getMember(userId);
-    return member?.rawDisplayName || userId || _t("Someone");
+    return member?.name || member?.rawDisplayName || userId || _t("Someone");
 }
```

Reply:
Maybe, it needs careful looking into as some places handle disambiguation themselves (e.g. the timeline where it is rendered in a different font size & colour)

Reply:
Looks like the only other place this function is used is

```
function textForMemberEvent(ev: MatrixEvent, allowJSX: boolean, showHiddenEvents?: boolean): () => string | null {
    // XXX: SYJS-16 "sender is sometimes null for join messages"
    const senderName = ev.sender?.name || getRoomMemberDisplayname(ev);
    const targetName = ev.target?.name || getRoomMemberDisplayname(ev, ev.getStateKey());
```

in the same file. So I think this change is safe due to the preceding `name ||`.